### PR TITLE
Add error-prone and NullAway

### PR DESCRIPTION
**Pull request by JakeWharton**
NullAway is still only set to warn, for now…

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -39,5 +52,18 @@ allprojects {
...
+    errorproneJavac 'com.google.errorprone:javac:9+181-r4173-1'
+    errorprone 'com.google.errorprone:error_prone_core:2.3.2'
+    errorprone 'com.uber.nullaway:nullaway:0.6.4'
+  }
+
+  tasks.withType(JavaCompile).configureEach {
+    options.errorprone {
+      disableWarningsInGeneratedCode = true
+      check("JavaxInjectOnAbstractMethod", CheckSeverity.OFF)
+      option("NullAway:AnnotatedPackages", "dagger.reflect")
```
Comment (TWiStErRob) [inline build.gradle]: reflect/src/main/java/dagger/Dagger.java
codegen/src/main/java/dagger/Dagger.java
codegen/src/main/java/dagger/internal/DaggerCodegen.java

File: build.gradle [hunk 2]
```diff
@@ -24,12 +26,23 @@ buildscript {
...
       'guava': 'com.google.guava:guava:27.0.1-jre',
   ]
+
+  repositories {
+    mavenCentral()
+    gradlePluginPortal()
+  }
+
+  dependencies {
+    classpath 'net.ltgt.gradle:gradle-errorprone-plugin:0.7'
```
Comment (TWiStErRob) [inline build.gradle]: no `ext.deps` on the new dependencies because they're only used once?

File: build.gradle (hunk 1 above)
Comment (JakeWharton) [reply build.gradle]: Those packages aren't owned by us so I don't want to include them and give a false sense of security

File: build.gradle (hunk 2 above)
Comment (JakeWharton) [reply build.gradle]: yes

**Pull request merged**